docs/variants/dell_optiplex/recovery.md

```diff
@@ -1,85 +1,210 @@
+`xac` is third file resulting from previous binary split, so it contains BIOS
+region which in case of Dell OptiPlex 9010/7010 should be flashed to 4MB SPI.
 
 ```bash
-flashrom -p internal:laptop=this_is_not_a_laptop -w /tmp/backup.rom -i me
+flashrom -w xac -p linux_spi:dev=/dev/spidev1.0,spispeed=16000 -c "MX25L3205D/MX25L3208D"
 ```
 
-If you have placed the jumper correctly, you should see the following message
-in flashrom's output:
+## (Optional) Step 7: Flash 8MB (ME) part
+
+When more serious problem with occur, like
+[broken ME](../faq/#cpu-was-replace-warm-reset-required-loop),
+it may be necessary to use your
+[firmware backup](../initial-deployment/#bios-backup)
+and restore content of 8MB chip.
+
+### Step 7a: Connect SOIC-16 Pomona clip between RTE and target
+
+Connect SOIC-16 Pomona according to
+[MX25L6406E datasheet](https://www.digikey.ch/htmldatasheets/production/980657/0/0/1/MX25L6406E.pdf).
+
+<center>
+![](../../images/mx25l6406e_pinout.png)
+</center>
+
+<center>
+ ![Pomona SOIC16 clip](../../images/soic16_pomona_clip.jpg)
+</center>
+
+<center>
+
+ | RTE J7                                 | Pomona SOIC clip  |
+ |:--------------------------------------:|:-----------------:|
+ | VCC                                    | pin 2 (upside)    |
+ | CS                                     | pin 7 (upside)    |
+ | MISO                                   | pin 8 (upside)    |
+ | SCLK                                   | pin 16 (downside) |
+ | MOSI                                   | pin 15 (downside) |
+ | GND                                    | pin 10 (downside) |
+
+</center>
+
+Numbers 1-8 have to be on one side and numbers 9-16 have to be on the other
+side of the clip.
+
+<center>
+![Clip up](../../images/soic16_clip_upside.jpg)
+</center>
+
+<center>
+![Clip down](../../images/soic16_clip_downside.jpg)
+</center>
+
+Clip on the `SPI_2/SPI_3` chip. Match pin 1 (`HOLD#`) of the Pomona clip with
+the first pin of `SPI_2/SPI_3` chip, marked with a small dot engraved on the
+chip.
+
+![Clip connected](../../images/soic16_clip_connected.jpg)
+
+### Step 7b: Flash 8MB (ME) part
 
 ```bash
-The Flash Descriptor Override Strap-Pin is set. Restrictions implied by
-the Master Section of the flash descriptor are NOT in effect. Please note
-that Protected Range (PR) restrictions still apply.
+echo 1 > /sys/class/gpio/gpio405/value
 ```
 
-A newer version of flashrom may not display the warning about unsupported
-chipset as it already may be marked as tested. Our team has verified that
-the flashrom updates firmware reliably on this chipset.
+```bash
+echo 1 > /sys/class/gpio/gpio406/value
+```
 
-If you will face any issues please refer to
-[troubleshooting section](https://docs.dasharo.com/variants/dell_optiplex/initial-deployment/#troubleshooting).
+```bash
+echo 1 > /sys/class/gpio/gpio404/value
+```
+
+```bash
+flashrom -w optiplex_flash_8M_IFD_ME_cleaned.bin -p linux_spi:dev=/dev/spidev1.0,spispeed=16000 -c "MX25L6406E/MX25L6408E"
```

Comment:
Again we must be careful, because we flash GBE region as well. This `optiplex_flash_8M_IFD_ME_cleaned.bin` has to be from a backup. We must specify the regions IFD and ME to flash and leave GBE untouched.

Reply:
@miczyg1 please suggest a command and I can test it. I did what was at hand at that time, and it was consulted with you on martix, but I understand we can do better.

Reply:
```suggestion
flashrom -w optiplex_flash_8M_IFD_ME_cleaned.bin -p linux_spi:dev=/dev/spidev1.0,spispeed=16000 -c "MX25L6406E/MX25L6408E --ifd -i fd -i me"
```

Let's try this for starters. Hopefully flashrom won't complain on layout that has addresses bigger than flash chip.